docs/manual/developer_guide.adoc

```diff
@@ -73,11 +76,22 @@ image::Patching-Pull_Requests.png[align="left"]
 image::Patching-Create_Pull.png[align="left"]
 
 4) You will be requested to input a patch title and description. Be concise, but thorough enough for a community member to understand logic behind your changes. Paste into the description field testing evidence (e.g. running testoval.py on any submitted OVAL, or before/after for remediation scripts).
+
+If you work on a feature or a bugfix that has an associated issue, either
+
+- assign yourself to the issue (if you have rights), or
+- mention the issue number in the pull request.
+
+This will improve the odds that multiple people won't work on a single issue without being aware of each other's work.
 After completing the form, select "Send pull request":
 
 image::Patching-Send_Pull_Request.png[align="left"]
 
-5) A community member will review your patch. They will either merge the patch upstream, indicate additional changes/documentation needed, or decline the patch. You'll automatically be notified via EMail of any status updates.
+5) Keep historical records clean - don't use git history-altering magic in your branch during your work on the pull request.
+If you e.g. squash commits, the pull request page will be broken - if you made some mistakes, got feedback and corrected those mistakes based on the feedback, nobody will be able to learn from your pull request, because commits introducing mistakes will disappear and comments of reviewers therefore won't make sense.
```

Comment:
Sorry, no. All commits should be clean if changes are on the content that was reviewed e.g. https://github.com/OpenSCAP/scap-security-guide/pull/2409 or https://github.com/OpenSCAP/scap-security-guide/pull/2418 unless you are adding new functionality/fixes in a commit e.g. https://github.com/OpenSCAP/scap-security-guide/pull/2003. PRs with commits like https://github.com/OpenSCAP/scap-security-guide/pull/2280/commits are a no go as it dirties up the git history making it hard to determine what the actual change was (examples of others I just found this one easily)

Reply:
Squashing sometimes makes sense but IMO it should be an exception and I would prefer if developers did it via `git rebase -i` to maybe squash some neighboring commits or reorganize it to be more understandable.

I am against blanket squashing on merge as a rule, I think it makes it harder to understand the code. Sometimes I run git blame to figure out what some of the lines mean, I want to see the commit and the commit message, I don't want to run into a huge commit.